lib/puppet_x/certs/provider/keystore.rb

```diff
@@ -24,6 +24,9 @@ def exists?
             if e.message.include?('java.security.UnrecoverableKeyException') || e.message.include?('keystore password was incorrect')
               Puppet.debug("Invalid password for #{store}")
               return false
+            elsif e.message.include?('Keystore file exists, but is empty')
+              Puppet.debug(e)
+              return false
             else
               Puppet.log_exception(e, "Failed to read keystore '#{store}'")
```

Comment:
Just wondering, should this also be
```suggestion
              Puppet.log_exception(e, "Failed to read keystore '#{store}'")
              return false
```

Reply:
I wasn't sure. Originally we said we do not know *what* exceptions could occur, and we should just halt execution for the user to inspect further rather than assuming. I could go either way still -- what do you think?

Reply:
That's a fair point. Let's keep it this way then